src/components/metrics/card/name/index.js

```diff
@@ -1,108 +1,117 @@
+   return this.setState({editorState})
+  }
+
+  focus() {
+    this.refs.editor.focus()
+  }
+
+  changePlainText(value) {
+    this.setState({editorState: this._plainTextEditorState(value)})
+  }
+
+  getPlainText() {
+    return this.state.editorState.getCurrentContent().getPlainText('')
+  }
+
+  render() {
+    const {editorState} = this.state;
+    return (
+      <div onClick={this.props.isClickable && this.focus.bind(this)}>
+        <Editor
+          editorState={editorState}
+          onBlur={this.props.onBlur}
+          onChange={this._onChange.bind(this)}
+          tabIndex={2}
+          ref='editor'
+          placeholder={this.props.placeholder}
+        />
+      </div>
+    );
+  }
+}
+
 export default class MetricName extends Component {
   displayName: 'MetricName'
 
   static propTypes = {
     name: PropTypes.string,
     inSelectedCell: PropTypes.bool.isRequired,
-    onChange: PropTypes.func.isRequired,
-    editable: PropTypes.bool.isRequired,
+    onChange: PropTypes.func.isRequired
   }
 
   state = {
-    value: this.props.name,
-    editing: false,
-    persistEditing: false,
-  }
-
-  _editable() {
-    if (this.state.persistEditing) {
-      return this.state.editing
-    } else {
-      return this.props.editable && this.state.editing
-    }
+    value: this.props.name
   }
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.name === this.state.value) {this.setState({value: nextProps.name})}
+    if (!nextProps.inSelectedCell && (this.props.name !== nextProps.name) && (nextProps.name !== this.value())) {
+      this.refs.SimpleEditor.changePlainText(nextProps.name)
+    }
   }
 
-  shouldComponentUpdate(nextProps, nextState) {
-    return ((nextProps.name !== this.props.name) ||
-            (nextProps.inSelectedCell !== this.props.inSelectedCell) ||
-            (nextProps.editable !== this.props.editable) ||
-            (nextState.value !== this.state.value) ||
-            (nextState.editing !== this.state.editing))
+  componentWillUnmount() {
+    this.handleSubmit()
   }
 
   handleSubmit() {
     if (this._hasChanged()){
-      this.props.onChange({name: this.state.value})
+      this.props.onChange({name: this.value()})
     }
-    this.setState({editing: false, persistEditing: false})
   }
 
   _hasChanged() {
-    return (this.state.value != this.props.name)
-  }
-
-  _handleMouseLeave() {
-    if (!(this.state.persistEditing || this._hasChanged())) {
-      this.setState({editing: false})
-    }
+    return (this.value() != this.props.name)
   }
 
   hasContent() {
-    return !_.isEmpty(this.state.value)
-  }
-
-  componentWillUnmount() {
-    this.handleSubmit()
+    return !_.isEmpty(this.value())
   }
 
-  onChange(e) {
-    this.setState({value: e.target.value})
+  value() {
+    return this.refs.SimpleEditor.getPlainText()
   }
 
   handleKeyDown(e) {
+    e.stopPropagation()
+    this.props.heightHasChanged()
     const ENTER = (e) => ((e.keyCode === 13) && !e.shiftKey)
```

Comment:
If you're going to keep it at least remove the stopPropagation line (which is unnecessary now) and add a comment that this is currently not working.

Side note, I don't know if we need this at all regardless, as tabbing works fine to jump between sections (and seems more intuitive to me. I would expect Enter to jump out of the cell entirely, perhaps, but not to jump between sections I don't think).